worlds/pokemon_emerald/__init__.py

```diff
@@ -306,10 +416,208 @@ def refresh_tm_choices() -> None:
                 self.multiworld.itempool.append(item)
 
     def set_rules(self) -> None:
+        from .rules import set_rules
         set_rules(self)
 
     def generate_basic(self) -> None:
```

Comment:
Yeah. This stuff has moved around quite a bit over the life of the apworld. At one point it was a pain to try to avoid circular dependencies, but it seems like the code has cleaned up enough that it's not such a hurdle any more.

Moved all these sections in `generate_basic` and `generate_output` into related helper files in [7ced289](https://github.com/ArchipelagoMW/Archipelago/pull/2918/commits/7ced28988811f0787a8f093cdf0c95a9d686446a). They're all pretty much exactly the same, only changing `self` to `world` and similarly trivial renames, potentially with an extra check to return early in case of a vanilla option value, to encapsulate a little bit of the surrounding code.

To make sure I didn't break anything in the move, I compared the md5s of a couple patched ROMs on the commits before and after the changes to make sure it produced the same results. Then I reordered the function calls a little for organization.